src/XRP/shared/gateway-balances.ts

```diff
@@ -0,0 +1,80 @@
+import { XrpUtils } from 'xpring-common-js'
+import { XrpError, XrpErrorType } from '.'
+import {
+  GatewayBalancesResponse,
+  CurrencyValuePair,
+} from './rippled-json-rpc-schema'
+
+/**
+ * Represents an amount of an Issued Currency.
+ */
+export class IssuedCurrencyValue {
+  readonly currency: string
+
+  readonly value: string
+
+  public constructor(currencyValuePair: CurrencyValuePair) {
+    this.currency = currencyValuePair.currency
+    this.value = currencyValuePair.value
+  }
+}
+
+/**
+ * Represents information about a gateway's issued currency balances on the XRPL, as of the most recent validated ledger version.
+ * @see https://xrpl.org/gateway_balances.html
+ */
+export default class GatewayBalances {
```

Comment:
same q for this guy - if you don't expect it to ever "do" anything, i think it's better to use an `interface`

since TS is a structural type system, creating a `class` implies that you have aspirations for the object to "do" stuff (a la OOP). if you instead make it an `interface` and construct it using a function (like `createGatewayBalance` or `initGatewayBalance` or whatever), it's super clear that this thing, particularly with readonly properties, represents a collection of data but isn't an actor itself

Reply:
dope went interfacey here too